libobs/obs-encoder.c

```diff
@@ -710,6 +837,29 @@ void obs_encoder_set_scaled_size(obs_encoder_t *encoder, uint32_t width,
 	encoder->scaled_height = height;
 }
 
+void obs_encoder_set_gpu_scale_type(obs_encoder_t *encoder,
```

Comment:
Would I be correct in assuming that this GPU scaling feature is for shared texture encoders only? If so, we should impose that restriction here accordingly.

Reply:
Nope, it's for all encoders, though I haven't done any benchmarking to see how big the difference is when using e.g. x264

Reply:
Might be worth doing some more testing on this later, because it does does pose a valid question about whether we should consider this for all encoders in the future.

I don't feel this should block the PR for the time being though because it might require a lot more testing and possibly more code. But definitely something to consider.